crane/search/base.py

```diff
@@ -64,6 +64,24 @@ def _filter_result(self, result):
             return False
         return True
 
+    def _filter_result_v2(self, result):
+        """
+        Determines if a given result object, which represents a repository, is
+        both known by this app (aka we have it in the app data), and if the user
+        is authorized to access it.
+
+        :param result:  one search result
+        :type  result:  SearchResult
+        :return:    True iff the repository is known and the user is authorized,
+                    else False
+        :rtype:     bool
+        """
+        try:
```

Comment:
what about keeping the just the _filter_result() and instead:

``` 
        try:
            app_util.name_is_authorized(result.name)
        except exceptions.HTTPError:
            try:
                app_util.repo_is_authorized(result.name)
            except exceptions.HTTPError:
                return False
        return True
```

or

```
       try:
            if  not app_util.name_is_authorized(result.name):
                return True
           app_util.repo_is_authorized(result.name)
        except exceptions.HTTPError:
            return False
        return True

This solution will simplify: https://github.com/pulp/crane/pull/81/files#diff-aaa7c890485705f9a4ef1adebdd8784dR47
You would not need to call the filter_result twice on the results.
The protection comes per repo and not per type/version of the content.

And you also would be able to get rid of https://github.com/pulp/crane/pull/81/files#diff-aaa7c890485705f9a4ef1adebdd8784dR105

Reply:
I agree that these should be a single function that does as many checks as necessary to get its job done. A great indicator is that the doc blocks are identical.